aten/src/ATen/core/Vitals.cpp

```diff
@@ -30,5 +33,26 @@ bool torchVitalEnabled() {
   return enabled;
 }
 
+bool APIVitals::setVital(
+    const std::string& vital_name,
+    const std::string& attr_name,
+    const std::string& value) {
+  if (!torchVitalEnabled()) {
+    return false;
+  }
+
+  auto iter = name_map_.find(vital_name);
+  TorchVital *vital = nullptr;
+  if (iter == name_map_.end()) {
```

Comment:
nit: instead of trying to find `vital_name`, you can just unconditionally insert it, and check if insertion was successful? Or is TorchVital(vital_name) constructor expensive?

Reply:
I like the how much cleaner the code would look. I think the issue is that TorchVital destructor prints to stdout; so we really don't want to call the destructor.